specs/Standards/Storage.md

```diff
@@ -0,0 +1,203 @@
+
+### Example scenarios
+
+#### Account pays for own storage
+
+Alice registers her account on a fungible token contract.
+
+**Assumptions**
+
+- Alice's account is `alice`.
+- The fungible token contract is `mochi`.
+- The fungible token contract stores account information that requires 2350000000000000000000 yoctoⓃ of storage.
+
+**High-level explanation**
+
+1. Alice checks to make sure she does not have storage on the `mochi` contract.
+2. Alice determines how much storage is needed to register an account with the `mochi` contract.
+3. Alice issues a transaction to deposit Ⓝ for her account.
+
+**Technical calls**
+
+1. Alice queries a view-only method to determine if she already has storage on this contract with `mochi::storage_balance_of({"account_id": "alice"})`. Using [NEAR CLI](https://docs.near.org/docs/tools/near-cli) to make this view call, the command would be:
+
+       near view mochi storage_balance_of '{"account_id": "alice"}'
+
+   The response:
+
+       View call: mochi.storage_balance_of({"account_id": "alice"})
+       { total: '0', available: '0' }
+
+2. Alice uses [NEAR CLI](https://docs.near.org/docs/tools/near-cli) to make a view call.
+
+       near view mochi storage_minimum_balance
+
+   The response:
+
+       '2350000000000000000000'
+
+   [2350000000000000000000 / 10^24](https://www.wolframalpha.com/input/?i=2350000000000000000000+%2F+10%5E24) (yocto) = 0.00235 Ⓝ
+
+3. Alice deposits the proper amount in a transaction by calling `mochi::storage_deposit` with the attached deposit of '0.00235'. Using NEAR CLI:
+
+       near call mochi storage_deposit '' --accountId alice --amount 0.00235
+
+   The result:
+
+       {
+         total: '2350000000000000000000',
+         available: '2350000000000000000000'
+       }
+
+
+#### Account pays for another account's storage
+
+Alice wishes to eventually send `MOCHI` tokens to Bob who is not registered. She decides to pay for Bob's storage.
+
+**Assumptions**
+
+- Alice's account is `alice`.
+- Bob's account is `bob`.
+- As in the prior example, the storage cost per user is 2350000000000000000000 yoctoⓃ or 0.00235 Ⓝ, and this is known to Alice.
+
+**High-level explanation**
+
+1. Alice issues a transaction to deposit Ⓝ for Bob's account.
+
+**Technical calls**
+
+1. Alice calls `mochi::storage_deposit({"account_id": "bob"})` with the attached deposit of '0.00235'. Using NEAR CLI the command would be:
+
+       near call mochi storage_deposit '{"account_id": "bob"}' --accountId alice --amount 0.00235
+
+    The result:
+
+       {
+         total: '2350000000000000000000',
+         available: '2350000000000000000000'
+       }
+
+#### Account withdraws storage deposit
+
+Alice and Bob decide to withdraw their storage deposit from the `mochi` contract.
+
+**Assumptions**
+
+- Alice's account is `alice`.
+- Bob's account is `bob`.
+- Alice and Bob's account balances are `0`.
```

Comment:
@oysterpack I've updated this whole example to avoid making it seem like this standard covers account removal/closure. Here's the updated version. Let me know if you have any feedback.

> #### Accounts withdraw excess storage deposit
> 
> Alice and Bob decide to withdraw some unused storage deposit from the `mochi` contract.
> 
> **Assumptions**
> 
> - Alice's account is `alice`.
> - Bob's account is `bob`.
> - Both Alice's and Bob's accounts have more deposited than they are using.
> 
> **High-level explanation**
> 
> 1. Knowing that she registered Bob's account and believing that this entitles her to receive a refund for that deposit, Alice views her & Bob's storage balances and sees each have extra.
> 2. Alice issues a transaction to withdraw her own excess deposit.
> 3. Alice attempts to withdraw Bob's excess deposit, but cannot.
> 4. Bob issues a transaction to withdraw his own excess deposit.
> 
> **Technical calls**
> 
> 1. Alice queries `mochi::storage_balance_of({ "account_id": "alice" })` and `mochi::storage_balance_of({ "account_id": "alice" })`.
> 
>    Checking her own account with NEAR CLI:
> 
>        near view mochi storage_balance_of '{"account_id": "alice"}'
> 
>    Response:
> 
>        View call: mochi.storage_balance_of({"account_id": "alice"})
>        {
>          total: '7050000000000000000000',
>          available: '2350000000000000000000'
>        }
> 
>    Checking Bob's account:
> 
>        near view mochi storage_balance_of '{"account_id": "bob"}'
> 
>    Response:
> 
>        View call: mochi.storage_balance_of({"account_id": "bob"})
>        {
>          total: '4700000000000000000000',
>          available: '2350000000000000000000'
>        }
> 
> 2. Alice calls `mochi::storage_withdraw({"amount": "2350000000000000000000"})` for her own account. NEAR CLI command:
> 
>        near call mochi storage_withdraw '{"amount": "2350000000000000000000"}' --accountId alice
> 
> 3. Alice realizes that `storage_withdraw` does not allow specifying the account to withdraw from. She has withdrawn all she can from her own account. When she re-checks `mochi::storage_balance_of({"account_id": "alice"})`, it indicates she has zero available balance. This is because storage withdrawal is only for the predecessor account that has signed the transaction. Alice cannot withdraw for Bob.
> 
> 4. Bob issues the same transaction as Alice did in step 2.
> 
>        near call mochi storage_withdraw '{"amount": "2350000000000000000000"}' --accountId bob

cc @mikedotexe

Reply:
I believe this does address account removal